addon/mixins/routes/application.js

```diff
@@ -31,6 +31,13 @@ export default Ember.Mixin.create({
         outlet: outlet,
         parentView: parentView
       });
+
+      // Check if another modal exists then give it focus
+      Ember.run.later(function() {
```

Comment:
When you use `Ember.run.later` it's intended to received a [third argument](http://emberjs.com/api/classes/Ember.run.html#method_later). I would say use run later as a last resort because it's kind of hacky here.